tests/session/bug1274-002.phpt

```diff
@@ -0,0 +1,102 @@
+--TEST--
+PHPC-1274: Session destruct should not end session from parent process
```

Comment:
This is currently not functioning as a regression test, as I'm having trouble getting the child process to actually kill the session when it exits if I disable the call to `mongoc_client_reset()` in the Session's `free_object` handler.

I think this may be because the session pool is really only destroyed during the call to `mongoc_client_destroy()`, and PHPC purposefully avoids calling that due to https://github.com/mongodb/mongo-php-driver/commit/600db56bd880e09502eb2e996cfd02c9aadc1ad3 ([PHPC-912](https://jira.mongodb.org/browse/PHPC-912)). If that's the case, perhaps we can't implement a regression test for this at all and will have to make do with the "does not abort" test.

Reply:
I'd like to leave this test in place despite my last comment, as it could become more relevant when we get to [PHPC-1522](https://jira.mongodb.org/browse/PHPC-1522). And even if we can't implement that (i.e. allow children to call `mongoc_client_destroy()` on clients created by parents), this may still be useful as our only test to demonstrate that child processes don't kill active sessions of a parent process.